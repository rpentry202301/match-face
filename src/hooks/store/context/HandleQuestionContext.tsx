"use client";

import React, { useContext } from "react";

type Refine = {
  word: string[];
  department: number[];
};

const SelectContext = React.createContext<[Refine, React.Dispatch<React.SetStateAction<Refine>>] | undefined>(undefined);

export function SelectProvider({ children }: { children: React.ReactNode }) {
  const [select, setSelect] = React.useState<Refine>({
    word: [],
    department: [],
  });

  return (
    <SelectContext.Provider value={[select, setSelect]}>
      {children}
    </SelectContext.Provider>
  );
}

export function useRefine() {
  const context = useContext(SelectContext);
  if (context === undefined) {
    throw new Error("useRefine must be used within a SelectProvider");
  }
  return context;
}
