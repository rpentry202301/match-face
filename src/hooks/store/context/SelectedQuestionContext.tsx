"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { QusetionsReducer } from "../reducer/selecterReducer";
import { QuestionsReducerAction } from "@/types/admin/tasks/register/types";

export type Questions = {
  projectId: number;
  list: {
    id: number;
    name: string;
  }[];
};

const SelectedQuestionContext = createContext<
  [Questions, Dispatch<QuestionsReducerAction>] | undefined
>(undefined);

export const SelectedQuestionProvider = ({
  children,
}: {
  children: ReactNode;
}) => {
  const [state, dispatch] = useReducer<
    Reducer<Questions, QuestionsReducerAction>
  >(QusetionsReducer, { projectId: 0, list: [] });
  return (
    <SelectedQuestionContext.Provider value={[state, dispatch]}>
      {children}
    </SelectedQuestionContext.Provider>
  );
};

export const useSelectedQuestion = () => {
  const context = useContext(SelectedQuestionContext);
  if (context === undefined)
    throw new Error("Function 'useSelectedQuestion' must be used in Provider.");
  return context;
};
