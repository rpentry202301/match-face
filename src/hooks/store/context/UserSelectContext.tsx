"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { UsersReducer } from "../reducer/selecterReducer";
import { UsersReducerAction } from "@/types/admin/tasks/register/types";

export type Users = {
  id: string;
  name: string;
};

const UserSelectContext = createContext<
  [Users[], Dispatch<UsersReducerAction>] | undefined
>(undefined);

export const UserSelectProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<Users[], UsersReducerAction>>(
    // selecterReducer,
    UsersReducer,
    []
  );
  return (
    <UserSelectContext.Provider value={[state, dispatch]}>
      {children}
    </UserSelectContext.Provider>
  );
};

export const useUserSelect = () => {
  const context = useContext(UserSelectContext);
  if (context === undefined)
    throw new Error("Function 'useUserSelect' must be used in Provider.");
  return context;
};
