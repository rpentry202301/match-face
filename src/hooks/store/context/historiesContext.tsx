'use client'
import { createContext, Dispatch, useContext, useReducer } from "react";
import { SelectHistoryAction } from "@/types/admin/histories/admin_histories";
import { SelectHistoryReducer } from "../reducer/historiesReducer";

const SelectHistoryContext = createContext<{month:string,department:number[],skills:number[]}| undefined>(undefined);
const SelectHistoryDispatchContext = createContext<Dispatch<SelectHistoryAction>|undefined>(undefined)
const initialSelect = {month:"",department:[],skills:[]}

export function SelectHistoryProvider ({children}:{children:React.ReactNode}){
    const [state,dispatch] = useReducer(SelectHistoryReducer,initialSelect)
    return (
        <SelectHistoryContext.Provider value={state}>
            <SelectHistoryDispatchContext.Provider value={dispatch} >
            {children}
            </SelectHistoryDispatchContext.Provider>
        </SelectHistoryContext.Provider>
    )
}

export const useSelectHistory = () => {
    const context = useContext(SelectHistoryContext)
    if(context === undefined) {
        throw new Error("Function 'useSelectHistory' must be used in Provider.")
    }
    return context
}

export const useSelectHistoryDispatch = () => {
    const dispatch = useContext(SelectHistoryDispatchContext)
    if(dispatch === undefined) {
        throw new Error("Function 'useSelectHistoryDispatch' must be used in Provider.")
    }
    return dispatch
}
