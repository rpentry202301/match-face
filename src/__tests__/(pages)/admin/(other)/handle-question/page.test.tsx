import { render, screen, fireEvent } from "@testing-library/react";
import HandleQuestionPage from "@/app/(pages)/admin/(other)/handle-question/page";
import React from "react";
import { SelectProvider } from "@/hooks/store/context/HandleQuestionContext";
import "@testing-library/jest-dom";

describe("質問・回答例一覧画面テスト", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    render(
      <SelectProvider>
        <HandleQuestionPage />
      </SelectProvider>
    );
  });
  afterAll(() => {
    jest.clearAllMocks();
  });

  it("スナップショット", async () => {
    const { container } = render(
      <SelectProvider>
        <HandleQuestionPage />
      </SelectProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("新規追加ボタンリンク", async () => {
    const createButton = screen.getByTestId("createButton");
    // fireEvent.click(createButton);
    expect(createButton).toHaveAttribute(
      "href",
      "/admin/handle-question/create"
    );
  });

  it("編集ボタンリンク", async () => {
    const editButton = screen.getByTestId("editButton_1");
    // fireEvent.click(editButton);
    expect(editButton).toHaveAttribute("href", "/admin/handle-question/edit/1");
  });

  it("Java絞り込み実行", async () => {
    const addWriteButton = screen.getByTestId("button-Java");
    const refineButton = screen.getByTestId("button-refine");
    const table = screen.getByTestId("projectTable");
    const descendant = screen.getByTestId("test2");
    const nonExistElement = screen.getByTestId("test1");
    fireEvent.click(addWriteButton);
    fireEvent.click(refineButton);
    //テーブル内にJava属性を持つ行が表示されているか
    expect(table).toContainElement(descendant);
    expect(descendant).not.toContainElement(table);
    //テーブル内にJava属性を持たない行が表示されていないか
    expect(table).not.toContainElement(nonExistElement);
  });

  it("PHP絞り込み&ワード検索実行", async () => {
    const addWriteButton = screen.getByTestId("button-PHP");
    const searchInput = screen.getByTestId("input-search");
    const refineButton = screen.getByTestId("button-refine");
    const table = screen.getByTestId("projectTable");
    const descendant = screen.getByTestId("test3");
    const nonExistElement = screen.getByTestId("test4");
    fireEvent.change(searchInput, { target: { value: "test3" } });
    fireEvent.click(addWriteButton);
    fireEvent.click(refineButton);
    //テーブル内にPHP属性かつtest3の行が表示されているか
    expect(table).toContainElement(descendant);
    expect(descendant).not.toContainElement(table);
    //テーブル内にPHP属性かつtest3を持たない行が表示されていないか
    expect(table).not.toContainElement(nonExistElement);
  });

  it("ワード検索実行", async () => {
    const searchInput = screen.getByTestId("input-search");
    const refineButton = screen.getByTestId("button-search");
    const table = screen.getByTestId("projectTable");
    const descendant = screen.getByTestId("test3");
    const nonExistElement = screen.getByTestId("test1");
    fireEvent.change(searchInput, { target: { value: "test3" } });
    fireEvent.click(refineButton);
    //テーブル内にtest3の行が表示されているか
    expect(table).toContainElement(descendant);
    expect(descendant).not.toContainElement(table);
    //テーブル内にtest3を持たない行が表示されていないか
    expect(table).not.toContainElement(nonExistElement);
  });

  it("ワード検索(複数部分一致)実行", async () => {
    const searchInput = screen.getByTestId("input-search");
    const refineButton = screen.getByTestId("button-search");
    const table = screen.getByTestId("projectTable");
    const descendant = screen.getByTestId("test1");
    const descendant2 = screen.getByTestId("test2");
    const nonExistElement = screen.getByTestId("test3");
    fireEvent.change(searchInput, { target: { value: "1 2" } });
    fireEvent.click(refineButton);
    //テーブル内にtest1とtest2の行が表示されているか
    expect(table).toContainElement(descendant);
    expect(table).toContainElement(descendant2);
    expect(descendant).not.toContainElement(table);
    expect(descendant2).not.toContainElement(table);
    //テーブル内に1と2を持たない行が表示されていないか
    expect(table).not.toContainElement(nonExistElement);
  });
});
