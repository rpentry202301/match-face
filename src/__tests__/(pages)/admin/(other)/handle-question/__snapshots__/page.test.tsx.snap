// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`質問・回答例一覧画面テスト スナップショット 1`] = `
<div>
  <div
    class="flex flex-col items-center my-10"
  >
    <div
      class="border-2 flex flex-col items-center w-5/12 max-w-5/12"
    >
      <div
        class="mt-7 flex items-center w-4/5 "
      >
        <div
          class=" w-4/5"
        >
          <div>
            <input
              class="border w-full pl-1 border-deep-gray"
              data-testid="input-search"
              id="search"
            />
          </div>
        </div>
        <div
          class=" w-1/5 flex justify-end"
        >
          <button
            class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm rounded-full border shadow-md text-xs py-1 px-5"
            data-testid="button-search"
          >
            検索
          </button>
        </div>
      </div>
      <div
        class="mt-7 flex w-4/5 justify-between"
      />
      <div>
        <button
          class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white p-2 h-auto rounded-none text-xs py-1 px-5 w-auto my-7 active:rounded-none active:text-xs active:py-1 active:px-5 active:w-auto active:my-7"
          data-testid="button-refine"
        >
          絞り込み
        </button>
      </div>
    </div>
    <a
      data-testid="createButton"
      href="/admin/handle-question/create"
      id="createButton"
    >
      <div>
        <button
          class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-lg my-10 active:my-10"
        >
          新規追加
        </button>
      </div>
    </a>
    <table
      class="w-4/5 border-2"
      data-testid="projectTable"
    >
      <tbody>
        <tr
          class="border-2 border-current bg-light-gray"
        >
          <th
            class="w-1/5 border-2 py-4"
          >
            最終編集日
          </th>
          <th
            class="w-1/5 border-2"
          >
            案件名
          </th>
          <th
            class="w-2/5 border-2"
          >
            案件概要
          </th>
          <th
            class="w-1/5 border-2"
          />
        </tr>
      </tbody>
    </table>
    <div
      class="w-5/12 max-w-5/12 flex justify-center my-7"
    >
      <ul
        class="flex"
      />
    </div>
  </div>
</div>
`;
